hashicorp/structure
# Deprecation - Remove multiple `Pds::***` components

### Description

Now that the main "app" layout (and sidebar navigation) have been completely refactored in Cloud UI we can finally start to remove a bunch of app-related components from Structure (and other components that were blocked by these components removal).

### Changes

In this PR I have 

- removed the following components/subcomponents (tests and documentation included)
  - `Pds::App` component with subcomponents
    - `Pds::App::Banner`
    - `Pds::App::Body`
    - `Pds::App::Drawer`
    - `Pds::App::Footer`
    - `Pds::App::Header`
    - `Pds::App::ModalArea`
    - `Pds::App::Sidebar`
  - `Pds::Button`
  - `Pds::CtaLink`
  - `Pds::Dropdown`
  - `Pds::Global[Header|Footer]`
  - `Pds::Nav`
  - `Pds::PageHeader`
  - `Pds::Page`

- added Storybook documentation files for the deprecated component
  - [App](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-app-deprecated--page)
  - [Button](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-button-deprecated--page)
  - [CtaLink](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-ctalink-deprecated--page)
  - [Dropdown](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-dropdown-deprecated--page)
  - [GlobalHeader/Footer](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-global-header-footer-deprecated--page)
  - [Nav](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-nav-deprecated--page)
  - [PageHeader](https://structure-git-remove-multiple-pds-components-hashicorp.vercel.app/?path=/story/deprecated-pageheader-deprecated--page)
  - [Page](http://localhost:6006/?path=/story/deprecated-page-deprecated--page)

### Testing

I have symlinked my local directory `package/pds-ember` to the `cloud-ui/node_modules/@hashicorp/pds-ember` folder in Cloud UI, run the application and tested that everything worked as expected.

Things I've found will need to be fixed, once the new Structure version is going to be adopted in Cloud UI:
- `packages/cloud-ui-core/addon/styles/components/clipboard-copy-button/index.scss`
  - the button is consuming the styles from the `Pds::Button`, we need to copy&paste them locally in the component
- `packages/cloud-ui-core/addon/styles/components/layout/sidebar-subnav-header.scss`
  - it's consuming the styles from the `nav/header`, we need to copy&paste them locally
  - notice: probably is not used anymore (to be checked with the Cloud UI engineers)
- `packages/cloud-ui-core/addon/styles/components/layout/sidebar-subnav-item.scss`
  - it's consuming the styles from the `nav/link`, we need to copy&paste them locally
  - notice: probably is not used anymore (to be checked with the Cloud UI engineers)
- `addons/data-visualizations/tests/dummy/app/templates/application.hbs`
  - still using `<Pds::App>` for testing purposes, needs to switch to the new page layout components
- other instances (in comments, storybook stories, etc) of other components, but we will need to look at things case by case and do a general cleanup of all the remnants of these deprecated components
### Relevant links
- https://hashicorp.atlassian.net/browse/HDS-1558
- [DS-038: Structure maintenance RFC](https://docs.google.com/document/d/1fr-fGPqx75IZgTWSHS8daX8IDHpXLZyEc-JHbGejSSU/edit)
